### PR TITLE
fix(ui) Fix displaying column level lineage for sibling nodes

### PR DESCRIPTION
**Pull request by chriscollins3456**
We had an issue where we sometimes were not displaying column level lineage properly for nodes that have siblings. This was happening because we had one "main" node that we display and if that node didn't have CLL, then we wouldn't show it. But one of its siblings may have CLL and we should be showing that if so.

This fixes that by creating a mapping of urns to inputs so we can create this `fineGrainedMap` (that's used to draw the lines between columns) and we check through each entity's siblings against this new mapping and update this `fineGrainedMap` to account for siblings and show CLL on the "visible" sibling node.

This ultimately this combines CLL between siblings and shows it as if it were one node.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/utils.ts [hunk 1]
```diff
@@ -146,3 +146,20 @@ export const handleBatchError = (urns, e, defaultMessage) => {
     }
     return defaultMessage;
 };
+
+// put all of the fineGrainedLineages for a given entity and its siblings into one array so we have all of it in one place
+export function getFineGrainedLineageWithSiblings(
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/entity/shared/utils.ts:151]: This is awesome!

**Review by aditya-radhakrishnan — APPROVED** [review]
Looks good! Do you have any before/after screenshots?

File: datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts [hunk 2]
```diff
@@ -42,24 +42,78 @@ function updateFineGrainedMap(
...
                         upstreamEntityUrn,
                         upstreamField,
-                        downstreamEntityUrn,
+                        lineageVizConfig.urn,
                         downstreamField,
                     );
+
+                    // upstreamEntityUrn could belong to a sibling we don't "render", so store its inputs to updateFineGrainedMap
+                    // and update the fine grained map later when we see the entity with these siblings
+                    // eslint-disable-next-line no-param-reassign
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts:70]: this is a bit challenging for me to grasp...

File: datahub-web-react/src/app/entity/EntityRegistry.tsx [hunk 3]
```diff
@@ -161,7 +166,8 @@ export default class EntityRegistry {
                     (genericEntityProperties?.upstream?.filtered || 0),
                 status: genericEntityProperties?.status,
                 siblingPlatforms: genericEntityProperties?.siblingPlatforms,
-                fineGrainedLineages: genericEntityProperties?.fineGrainedLineages,
+                fineGrainedLineages,
+                siblings: genericEntityProperties?.siblings,
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/EntityRegistry.tsx:170]: where is this being used?

File: datahub-web-react/src/app/lineage/types.ts [hunk 4]
```diff
@@ -51,6 +52,7 @@ export type FetchedEntity = {
     status?: Maybe<Status>;
     siblingPlatforms?: Maybe<DataPlatform[]>;
     fineGrainedLineages?: [FineGrainedLineage];
+    siblings?: Maybe<SiblingProperties>;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/lineage/types.ts:55]: is this being used on line 83 in constructtree?

File: datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts (hunk 2 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts:70]: Okay got it. Yeah makes sense - seems like siblings + graph is a really tricky combo. As long as this doesn't introduce any regression I'm happy! :)

Comment (chriscollins3456) [reply datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts:70]: haha for sure! things are looking good locally for me but i'll make sure to give it a once more around before merging.

**Pull request merged**